ts/components/screens/ScreenContentHeader.tsx

```diff
@@ -36,34 +31,69 @@ const styles = StyleSheet.create({
   }
 });
 
+const collapseTh = (0 as unknown) as Animated.AnimatedInterpolation;
+const elapseTh = (HEADER_HEIGHT as unknown) as Animated.AnimatedInterpolation;
+
 export class ScreenContentHeader extends React.PureComponent<Props> {
+  private heightAnimation: Animated.Value;
+  private elapse: Animated.CompositeAnimation;
+  private collapse: Animated.CompositeAnimation;
+
+  constructor(props: Props) {
+    super(props);
+
+    // Initialize animated value
+    this.heightAnimation = new Animated.Value(HEADER_HEIGHT);
+
+    // Animation to elapse the header height from 0 to HEADER_HEIGHT
+    this.elapse = Animated.timing(this.heightAnimation, {
+      toValue: HEADER_HEIGHT,
+      duration: 200
+    });
+
+    // Animation to collapse the header height from HEADER_HEIGHT to 0
+    this.collapse = Animated.timing(this.heightAnimation, {
+      toValue: 0,
+      duration: 200
+    });
+  }
+
+  public componentDidUpdate(prevProps: Props) {
+    if (this.props.dynamicHeight !== prevProps.dynamicHeight) {
```

Comment:
imho this is like a trick. This is what we do, let me know if I am wrong:

- listen from scroll event
- put scroll position in animated value
- interpolate that animated value
- use the result of interpolation to know if we have to expand or collapse the header

At the moment I can't find a cleaner solution. Anyway I did an attempt to simplify the code and the logic.
c7bee6e44295b605abff5447eea92cbe672709d2 (ignore test variable). This solution use the old interpolation to emit a value from 0 to 1: 
1-> header should collapse 
0-> header should expand
The logic is the same but this use 0/1 as ON/OFF values to trigger the animations

I found an interesting Animated class that could work in this scenario https://reactnative.dev/docs/animated#diffclamp

Reply:
your proposal added within https://github.com/pagopa/io-app/pull/1598/commits/1e3a6d889bb22f0aed2d59c104626eec6eb2b697. The method is interesting, I'll point it on a new story  (I'm sure something smarter exists for the desired behavior)